tests/test_templates_state_preparations.py

```diff
@@ -0,0 +1,218 @@
+            [0, 1, 2],
+            [-0.29972867+0.04964242j, -0.28309418+0.09873227j, 0.00785743-0.37560696j, -0.3825148 +0.00674343j, -0.03008048+0.31119167j, 0.03666351-0.15935903j, -0.25358831+0.35461265j, -0.32198531+0.33479292j],
+        ),
+        (
+            [-0.39340123+0.05705932j, 0.1980509 -0.24234781j, 0.27265585-0.0604432j, -0.42641249+0.25767258j, 0.40386614-0.39925987j, 0.03924761+0.13193724j, -0.06059103-0.01753834j, 0.21707136-0.15887973j],
+            [0, 1, 2],
+            [-0.39340123+0.05705932j, 0.1980509 -0.24234781j, 0.27265585-0.0604432j, -0.42641249+0.25767258j, 0.40386614-0.39925987j, 0.03924761+0.13193724j, -0.06059103-0.01753834j, 0.21707136-0.15887973j],
+        ),
+        (
+            [-1.33865287e-01+0.09802308j, 1.25060033e-01+0.16087698j, -4.14678130e-01-0.00774832j, 1.10121136e-01+0.37805482j, -3.21284864e-01+0.21521063j, -2.23121454e-04+0.28417422j, 5.64131205e-02+0.38135286j, 2.32694503e-01+0.41331133j],
+            [0, 1, 2],
+            [-1.33865287e-01+0.09802308j, 1.25060033e-01+0.16087698j, -4.14678130e-01-0.00774832j, 1.10121136e-01+0.37805482j, -3.21284864e-01+0.21521063j, -2.23121454e-04+0.28417422j, 5.64131205e-02+0.38135286j, 2.32694503e-01+0.41331133j],
+        ),
+        ([1/2, 0, 0, 0, 1j/2, 0, 1j/math.sqrt(2), 0], [0, 1, 2], [1/2, 0, 0, 0, 1j/2, 0, 1j/math.sqrt(2), 0]),
+        ([1/2, 0, 1j/2, 1j/math.sqrt(2)], [0, 1], [1/2, 0, 0, 0, 1j/2, 0, 1j/math.sqrt(2), 0]),
+    ])
+    # fmt: on
+    def test_state_preparation(self, tol, qubit_device_3_wires, state_vector, wires, target_state):
+        """Tests that the template MottonenStatePreparation integrates correctly with PennyLane."""
+
+        @qml.qnode(qubit_device_3_wires)
+        def circuit():
+            MottonenStatePreparation(state_vector, wires)
+
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))
+
+        circuit()
+
+        state = qubit_device_3_wires._state
+        fidelity = abs(np.vdot(state, target_state))**2
+
+        # print("state: ", state)
+        # print("target: ", target_state)
+        # print("fidelity: ", fidelity)
+
+        # We test for fidelity here, because the vector themselves will hardly match
+        # due to imperfect state preparation
+        assert np.isclose(fidelity, 1, atol=tol, rtol=0)
+
+    # fmt: off
+    @pytest.mark.parametrize("state_vector,wires,error_message", [
+        ([0, 1, 0], [0, 1], "Number of entries in the state vector must be equal to 2 to the power of the number of wires"),
+        ([0, 1, 0, 0, 0], [0], "Number of entries in the state vector must be equal to 2 to the power of the number of wires"),
+        ([0], 0, "Wires must be passed as a list of integers"),
+        ([1/2, 1/2], [0], "State vector probabilities have to sum up to 1.0"),
+        ([2/3, 0, 2j/3, -2/3], [0, 1], "State vector probabilities have to sum up to 1.0"),
+    ])
```

Comment:
This is a matter of style, but I find this very hard to read. Ideally, this should likely be separate atomic tests (e.g., `test_invalid_wires_type`, `test_invalid_state` etc.)

Reply:
Will do!